swan-lake/development-tutorials/source-code-dependencies/customizing-formatting.md

```diff
@@ -0,0 +1,389 @@
+---
+title: Customization of formatting
+description: The section on providing custom configurations to the formatter.
+keywords: ballerina, programming language, formatting, custom configurations
+permalink: /learn/customizing-formatting/
+active: customizing-formatting
+intro: This feature allows developers to customize the formatting of Ballerina projects. This is done with the use of a configuration file. This configuration file can be a local or a remote file, which allows for consistency in code style across projects and simplifies the process of enforcing formatting standards. This is introduced as an experimental feature in Ballerina 2201.9.0.
+---
+
+## Configuring
+
+Users can provide custom formatting configurations with the use of a separate configuration file of the `toml` format. Users can provide an arbitrary name for the file.
+
+The formatting configuration file can be specified in two ways:
+
+* `Ballerina.toml`: Define the path to a configuration file within the `Ballerina.toml` itself. The path can be absolute, relative to the package root, or a URL (HTTP/HTTPS). This path is specified under the `format` section in `Ballerina.toml` with the use of the `configPath` attribute.
+
+    ```
+    [format]
+    configPath = "Format.toml"
+    ```
+
+* `Format.toml`: If no configuration path is specified in `Ballerina.toml`, the formatter will look for a file named `Format.toml` in the current package root directory.
+
+If the `configPath` argument is not provided in the `Ballerina.toml` or the `Format.toml` file does not exist in the project root, default formatting options will be applied.
+
+If the format `configPath` is an HTTP/HTTPS URL, a cache of the configuration file will be maintained inside the `target` directory. If the remote configuration file is modified, then make sure to delete the cached `Format.toml` file inside the `target/format/` directory.
```

Comment:
```suggestion
If the format `configPath` value is an HTTP/HTTPS URL, the configuration file will be cached in the `target` directory. If the remote configuration file is modified, then make sure to delete the cached `Format.toml` file inside the `target/format/` directory.
```

Did we discuss this user experience anywhere?

Reply:
We did have a discussion on this in the earlier stages of the design and decided that it is a responsibility of the user to remove the cached file if it is updated